New marker conversion helpers (#7602)
ckeditor/ckeditor5: docs/framework/guides/architecture/editing-engine.md

* They can only be created and changed through the {@link module:engine/model/writer~Writer model writer}.
* They can be synchronized over the network with other collaborating clients.
* They are automatically updated when the document's structure is changed.

> * They can be {@link module:engine/conversion/downcasthelpers~DowncastHelpers#markerToHighlight converted to the editing view}, to show them in the editor.

Reviewer: What about `markerToElement()`? Shouldn't we mention both things:

```
	* They can be converted to the editing view, to show them in the editor (as {@link module:engine/conversion/downcasthelpers~DowncastHelpers#markerToHighlight highlights} or {@link module:engine/conversion/downcasthelpers~DowncastHelpers#markerToElement elements).
```

Author: For a moment that was deprecated so I didn't include it. But since it will not be deprecated after all, we may mention it.

* They can be {@link module:engine/conversion/downcasthelpers~DowncastHelpers#markerToData converted to the data view}, to store them with the document data.
* They can be {@link module:engine/conversion/upcasthelpers~UpcastHelpers#dataToMarker loaded with the document data}.

> Markers are ideal for storing and maintaining additional data &mdash; such as comments or selections of other users &mdash; that is related to some portion of the content.

Reviewer: ```suggestion
Markers are ideal for storing and maintaining additional data related to portions of the document &mdash; such as comments or selections of other users.
```
